### PR TITLE
Fix No Feedback Defaults to 0 Score (Negative Feedback) in Snowflake

### LINKED ISSUE #275: Add Metric to Track Usage Source (Slack Bot vs Webapp)
**Please describe the feature you'd like to see**
Add Metric to Track Usage Source (Slack Bot vs Webapp)

**Describe the solution you'd like**
Add metric tracking so that we can see how many/what proportion of requests are from slack app vs webapp, and specifically be able to trace individual requests to see which one is from where

**Are there any alternatives to this feature?**
Perhaps we can also track specifically which slack server this is used, or other details that maybe useful.

### LINKED ISSUE #291: Load Firestore to Snowflake DAG Defaulting to 0 Score when No Feedback
**Describe the bug**

**Version**
* OS: 

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### LINKED ISSUE #296: [Blocked] Prod and Dev Environments Sharing Same Fire Store Collections/Indexes
**Describe the bug**
There's an issue with sharing the same Firestore collections and indexes between the production and development environments. Currently, both environments are using the same Firestore setup, causing conflicts and data integrity issues. The metrics being tracked and put into snowflake is also incorrect because of this

Our me

**To Reproduce**
1. Access the Ask-Astro website or Ask-Astro Slackbot in the production environment.
2. Ask questions/send requests
3. Repeat the same actions in the development environment.
4. The same Firestore collections/indexes being used

**Expected behavior**
The production and development environments should have separate Firestore collections and indexes to ensure data isolation and prevent interference between the two environments.

**Screenshots**
N/A

### PR DESCRIPTION
**Pull request by davidgxue**
### Description
- Super simple code change
- Also changing snowflake db table schema (not code change not shown in PR)
- Tested by creating new firestore collection indexes

closes #291
closes #296 (second issue is used to test, no code change related)
closes #275

### CODE REVIEW

**Review by pankajastro — APPROVED** [review]

**Review by pankajkoti — APPROVED** [review]

File: airflow/dags/metrcis/load_firestore_to_snowflake.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import os
 import time
 from datetime import date, datetime, timedelta
+from typing import Optional
```
Comment (Lee-W) [inline airflow/dags/metrcis/load_firestore_to_snowflake.py:9]:
```suggestion
```

File: airflow/dags/metrcis/load_firestore_to_snowflake.py [hunk 2]
```diff
@@ -54,11 +55,11 @@ def load_request_data_from_firestore() -> list[tuple[str, int, bool, int]]:
             .stream()
         )
 
-        rows: list[tuple[str, int, bool, int]] = []
+        rows: list[tuple[str, Optional[int], bool, datetime]] = []
```
Comment (Lee-W) [inline airflow/dags/metrcis/load_firestore_to_snowflake.py:58]:
```suggestion
        rows: list[tuple[str, int | None, bool, datetime]] = []
```

**Review by Lee-W — APPROVED** [review]

**Review by pankajastro — APPROVED** [review]

**Review by jlaneve — APPROVED** [review]

**Review by sunank200 — CHANGES_REQUESTED** [review]
Can you add the test cases for the change

**Comment by davidgxue** [comment]
> Can you add the test cases for the change

There was no unit test for this original file to begin with. In fact, there isn't a single unit test for the entire airflow code when the previous team worked on this. I can start an initiative for better test coverage to add them, but due to the urgency of this bug fix PR, I don't think it makes sense to write all the unit tests here.

**Review by sunank200 — APPROVED** [review]

**Pull request merged**